Sources/SwiftParser/Expressions.swift

```diff
@@ -510,6 +527,21 @@ extension Parser {
       )
     }
 
+    // Try parse an 'if' or 'switch' as an expression. Note we do this here in
+    // parseUnaryExpression as we don't allow postfix syntax to hang off such
+    // expressions to avoid ambiguities such as postfix '.member', which can
+    // currently be parsed as a static dot member for a result builder.
+    if self.at(.keyword(.switch)) {
+      return RawExprSyntax(
+        parseSwitchExpression(switchHandle: .constant(.keyword(.switch)))
+      )
+    }
+    if self.at(.keyword(.if)) {
+      return RawExprSyntax(
+        parseIfExpression(ifHandle: .constant(.keyword(.if)))
+      )
+    }
+
```

Comment:
We could probably unify this with `repeat` above into a `switch`

```suggestion
    // Try parse an 'if' or 'switch' as an expression. Note we do this here in
    // parseUnaryExpression as we don't allow postfix syntax to hang off such
    // expressions to avoid ambiguities such as postfix '.member', which can
    // currently be parsed as a static dot member for a result builder.
    switch self.currentToken.rawTokenKind {
      // Same for `repeat`
      case .keyword(.switch):
        return RawExprSyntax(
          parseSwitchExpression(switchHandle: .constant(.keyword(.switch)))
        )
      case .keyword(.if):
        return RawExprSyntax(
          parseIfExpression(ifHandle: .constant(.keyword(.if)))
        )
```